packages/shared/src/hooks/useDefaultFeed.ts

```diff
@@ -1,14 +1,53 @@
+import { useEffect } from 'react';
+import { MainFeedPage } from '../components/utilities';
 import usePersistentContext from './usePersistentContext';
 
-export default function useDefaultFeed(): [
-  string,
-  (value: string) => Promise<void>,
-] {
+interface UseDefaultFeedProps {
+  hasFiltered?: boolean;
+  hasUser?: boolean;
+  feed?: string;
+}
+
+export const getShouldRedirect = (
+  isOnMyFeed: boolean,
+  isLoggedIn: boolean,
+): boolean => {
+  if (!isOnMyFeed) {
+    return false;
+  }
+
+  if (!isLoggedIn) {
+    return true;
+  }
+
+  return false;
+};
+
+export default function useDefaultFeed({
+  hasFiltered,
+  hasUser,
+  feed,
+}: UseDefaultFeedProps = {}): string {
+  const isMyFeed = feed === MainFeedPage.MyFeed;
   const [defaultFeed, updateDefaultFeed] = usePersistentContext(
     'defaultFeed',
-    'my-feed',
+    !hasUser || !hasFiltered ? 'popular' : 'my-feed',
     ['my-feed', 'popular', 'upvoted', 'discussed'],
+    !hasUser || !hasFiltered ? 'popular' : 'my-feed',
   );
 
-  return [defaultFeed, updateDefaultFeed];
+  useEffect(() => {
+    if (
+      defaultFeed !== null &&
+      feed !== null &&
+      feed !== defaultFeed &&
+      feed !== 'default' &&
+      feed !== '/' &&
```

Comment:
Suggestion to make it more readable:

```
![null, defaultFeed, 'default', '/'].includes(feed)
```

Reply:
I think we should use `every` as includes will make the condition act like an `OR`. Will refactor this one.